usecase/session.go

```diff
@@ -15,22 +15,43 @@ var syncCheckOffset = 5 * time.Second
 
 // SessionUseCase はセッションに関するユースケースです。
 type SessionUseCase struct {
-	tm        *entity.SyncCheckTimerManager
-	repo      repository.Session
-	playerCli spotify.Player
-	pusher    event.Pusher
+	tm          *entity.SyncCheckTimerManager
+	sessionRepo repository.Session
+	userRepo    repository.User
+	playerCli   spotify.Player
+	pusher      event.Pusher
 }
 
 // NewSessionUseCase はSessionUseCaseのポインタを生成します。
-func NewSessionUseCase(repo repository.Session, playerCli spotify.Player, pusher event.Pusher) *SessionUseCase {
+func NewSessionUseCase(sessionRepo repository.Session, userRepo repository.User, playerCli spotify.Player, pusher event.Pusher) *SessionUseCase {
 	return &SessionUseCase{
-		tm:        entity.NewSyncCheckTimerManager(),
-		repo:      repo,
-		playerCli: playerCli,
-		pusher:    pusher,
+		tm:          entity.NewSyncCheckTimerManager(),
+		sessionRepo: sessionRepo,
+		userRepo:    userRepo,
+		playerCli:   playerCli,
+		pusher:      pusher,
 	}
 }
 
+// CreateSession は与えられたセッション名のセッションを作成します。
+func (s *SessionUseCase) CreateSession(sessionName string, creatorID string) (*entity.SessionWithUser, error) {
+	creator, err := s.userRepo.FindByID(creatorID)
+	if err != nil {
+		return nil, fmt.Errorf("CreateSession sessionName=%s: %w", sessionName, err)
+	}
+
+	newSession, errNewSession := entity.NewSession(sessionName, creatorID)
+	if errNewSession != nil {
+		return nil, fmt.Errorf("CreateSession sessionName=%s: %w", sessionName, errNewSession)
+	}
```

Comment:
最初の `err` の情報を保持しておく必要はないので 、シンプルにerrという名前で良いかなと思った！

後、エラーメッセージは「何を呼び出したときにエラーになったのか」を分かるようにしたいので、呼び出している関数名をエラーメッセージに含めたほうがデバッグがしやすいかな〜

```suggestion
	newSession, err := entity.NewSession(sessionName, creatorID)
	if err != nil {
		return nil, fmt.Errorf("NewSession sessionName=%s: %w", sessionName, err)
	}
```